### ISSUE TITLE
mongodb starts without --replSet option on ubuntu

### ISSUE THREAD

**Issue by krukgit** [edelight/chef-mongodb#169]
Replica sets don't work under ubuntu 12.04 because mongod process starts without --replSet option.

I fixed it temporarily by appending the following to my attributes file:

```
default['mongodb']['sysconfig']['DAEMON_OPTS'] = "--config #{node['mongodb']['configfile']} --replSet #{node['mongodb']['replicaset_name']}"
```

It also was not entirely clear to me if node[:mongodb][:replicaset_name] is necessary to set up replicas.

**Comment by jamesonjlee** [comment]
you can set the replSet in the configuration file by doing `default['mongodb']['config']['replSet'] = setname`, some of the replicaset stuff does need to get moved around again.

**Comment by msaffitz** [comment]
+1 on amazon linux as well.  

It looks a lot like this broke here:  https://github.com/edelight/chef-mongodb/commit/34a3df1421dd849c767d7676face7e4c7104f2c5

It seems like it now expects the replica set name to be manually set (https://github.com/edelight/chef-mongodb/commit/15c20a6766bff04fade9caf7d7014a33462974dc#diff-d43736499048c4314212798e1c7f5093R21)

It used to be derived from the cluster name (https://github.com/edelight/chef-mongodb/blob/master/definitions/mongodb.rb#L52)

**Comment by willejs** [comment]
+1 on this. can the maintainer please start merging these fixes?

**Comment by josephholsten** [comment]
@willejs is there a particular PR you're referring to? I'll gladly review.

**Comment by josephholsten** [comment]
oh, I've combined #171's fix for this into #207, that should get in soon

**Comment by kikicarbonell** [comment]
:+1:

**Comment by jamesonjlee** [comment]
was merged in via #207 please report if you are still having this issue @krukgit

**Comment by msaffitz** [comment]
@jamesonjlee I'm still seeing this issue on Amazon linux with fd7ab07ebfd54fc1049bc37367b92842f37310ac

**Comment by jamesonjlee** [comment]
@msaffitz as in setting the replSet in the options?

**Comment by msaffitz** [comment]
yeah-- the only way to get the replSet in the mongod config set is to explicitly include `default['mongodb']['config']['replSet'] = setname` somewhere in the chef config.  We're using a sharded cluster, and used to just provide `default['mongodb']['shard_name']` and let the cookbook fallback to the default replicaset name (e.g. it would add an `rs_` prefix).  The new behavior is OK, but at the least the readme docs for a sharded replicaset should be updated.
